topic7/contracts/Membership.sol

```diff
@@ -0,0 +1,106 @@
+pragma solidity ^0.4.23;
+
+import './Destructible.sol';
+import './SafeMath.sol';
+
+contract Membership is Destructible {
+    
+    using SafeMath for uint;
+
+    uint public memberCount = 0;
+
+    struct Member {
+        bool isMember;
```

Comment:
Can we avoid keeping that variable in the storage? If the `lastDonationTimestamp` is initialized(is not zero), doesn't that mean that the address is member? Also this will allow the EVM to pack tightly the arguments of the struct, check this reading - [link](http://solidity.readthedocs.io/en/v0.4.24/miscellaneous.html#layout-of-state-variables-in-storage)

Reply:
I can't think of a way to avoid the variable. I went and checked the softuni solution and they also used a storage variable. Then I checked your solution and you used the same variable in a struct; is that not the same thing? As for the lastDonationTimestamp - fixed.

Reply:
Nah, buddy, I didn't use boolean variable in my solution, here is my struct: 
> 	struct MemberDetails {
		uint256 totalDonations;
		uint256 lastDonationTime;
		uint256 lastDonationAmount;
	}

And here is how I check:

> 	function isMember(Data storage _self, address _memberToCheck) internal view returns (bool) {
		return _self.members[_memberToCheck].lastDonationTime != 0;
	}

Guess my idea is kind of confusing, which is ok.
When it comes to `lastDonationTimestamp` what did you fix?

Reply:
Slight misunderstanding. Everything is fine ;)